src/main/java/io/jenkins/plugins/sprp/PipelineSnippetGenerator.java

```diff
@@ -138,30 +144,44 @@ public String getStage(
         if(stage.getFailure() != null
                 || stage.getSuccess() != null
                 || stage.getAlways() != null
-                || (stage.getName().equals("Build") && (archiveArtifacts != null || buildResultPaths != null))
-                || stage.getName().equals("Tests") && testResultPaths != null) {
+                || (stage.getName().equals("Build") &&
+                        (archiveArtifacts != null || buildResultPaths != null || findbugs != null))
+                || stage.getName().equals("Tests") && (testResultPaths != null || gitConfig.getGitUrl() != null)) {
             snippet += "\tpost {\n";
 
             if (stage.getSuccess() != null
-                    || (stage.getName().equals("Build") && (archiveArtifacts != null || buildResultPaths != null))
-                    || stage.getName().equals("Tests") && testResultPaths != null) {
+                    || (stage.getName().equals("Build"))
+                    || stage.getName().equals("Tests") && (testResultPaths != null || gitConfig.getGitUrl() != null)
+                    )
+            {
                 snippet += "\t\tsuccess {\n";
                 if (stage.getName().equals("Build")) {
+                    snippet += "\t\t\t" + addTabs("archiveArtifacts artifacts: '**/target/*.jar'\n", 3);
```

Comment:
Small comment: do you want to autogenerate the tabbing?  It looks like there are a few different ways that the number of tabs are inserted into the string.  I'd try to keep it consistent.

Reply:
I will open a Jira ticket to find ways to auto-generate tabbing and to make it consistent.